headless/src/main.rs

```diff
@@ -1,58 +1,91 @@
-    sim_flags.opts.enable_pandemic_model = Some(XorShiftRng::from_seed([sim_flags.rng_seed; 16]));
     // Less spam
     sim_flags.opts.alerts = AlertHandler::Silence;
     let mut timer = Timer::new("setup headless");
-    let (mut map, _, mut rng) = sim_flags.load(&mut timer);
-    map.hack_override_offstreet_spots(num_days);
-    // After the map changes, have to re-create the (empty) Sim, because things like
-    // ParkingSimState are out of sync.
-    let mut sim = Sim::new(&map, sim_flags.opts.clone(), &mut timer);
-
-    let base_scenario: Scenario = abstutil::read_binary(
-        abstutil::path_scenario(map.get_name(), "weekday"),
-        &mut timer,
-    );
-    ScenarioModifier::RepeatDays(num_days)
-        .apply(&map, base_scenario, &mut rng)
-        .instantiate(&mut sim, &map, &mut rng, &mut timer);
-    timer.done();
+    let (map, sim, _) = sim_flags.load(&mut timer);
+    *MAP.write().unwrap() = map;
+    *SIM.write().unwrap() = sim;
 
-    run_experiment(&map, &mut sim);
+    let addr = std::net::SocketAddr::from(([127, 0, 0, 1], port));
+    println!("Listening on http://{}", addr);
+    let serve_future = Server::bind(&addr).serve(hyper::service::make_service_fn(|_| async {
+        Ok::<_, hyper::Error>(hyper::service::service_fn(serve_req))
+    }));
+    if let Err(err) = serve_future.await {
+        panic!("Server error: {}", err);
+    }
 }
 
-fn run_experiment(map: &Map, sim: &mut Sim) {
-    let timer = Timer::new("run sim until done");
-    sim.run_until_done(
-        &map,
-        |sim, _map| {
-            // This'll run every 30 sim seconds
-            if false {
-                println!(
-                    "At {}, {} infected",
-                    sim.time(),
-                    prettyprint_usize(sim.get_pandemic_model().unwrap().count_infected())
-                );
-            }
-        },
-        None,
+async fn serve_req(req: Request<Body>) -> Result<Response<Body>, hyper::Error> {
+    let path = req.uri().path();
+    // Scrape query parameters
+    let params: HashMap<&str, &str> = req
+        .uri()
+        .query()
+        .unwrap_or("")
+        .split("&")
+        .filter(|x| !x.is_empty())
+        .map(|kv| {
+            let parts = kv.split("=").collect::<Vec<_>>();
+            // TODO Error handling
+            assert_eq!(parts.len(), 2);
+            (parts[0], parts[1])
+        })
+        .collect();
+
+    let resp = handle_command(
+        path,
+        params,
+        &mut SIM.write().unwrap(),
+        &MAP.read().unwrap(),
     );
-    timer.done();
-    println!("Done at {}", sim.time());
+    Ok(Response::new(Body::from(resp)))
+}
+
+fn handle_command(path: &str, params: HashMap<&str, &str>, sim: &mut Sim, map: &Map) -> String {
+    match path {
+        "/get-time" => sim.time().to_string(),
+        "/goto-time" => {
+            let t = Time::parse(params["t"]).unwrap();
+            if t <= sim.time() {
```

Comment:
HTTP 422 error? 

but meh, seems like we're not super invested in this API being super robust at this point, so take it or leave it.

Reply:
Yeah, I'll put in effort here once there's a better idea of what this API should do. I glanced through more frameworky options for serving an API. Maybe we should use protobuf, thrift, jsonrpc, something else.